classes/migration/upgrade/v3_5_0/I10157_CreateAndMigrateIssueIdField.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace APP\migration\upgrade\v3_5_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class I10157_CreateAndMigrateIssueIdField extends \PKP\migration\Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // add the new column to publications table
+        Schema::table('publications', function (Blueprint $table) {
+            $table->bigInteger('issue_id')->nullable()->after('submission_id');
+        });
+
+        // add the foreign key constraint and index,
+        Schema::table('publications', function (Blueprint $table) {
+            $table->foreign('issue_id')
+                ->references('issue_id')
+                ->on('issues')
+                ->nullOnDelete();
+            $table->index(['issue_id'], 'publications_issue_id_index');
+        });
+
+        // migrate data from publication_settings to the new issue_id column.
+        if (DB::getDriverName() === 'pgsql') {
```

Comment:
Lately we've been relying on code more like...
```php
use Illuminate\Database\MariaDbConnection;
use Illuminate\Database\MySqlConnection;
use Illuminate\Database\PostgresConnection;

...

switch (true) {
    case DB::connection() instanceof PostgresConnection:
        // Do a PostgreSQL query
        break;
    case DB::connection() instanceof MySqlConnection:
    case DB::connection() instanceof MySqlConnection:
        // Do a MySQL/MariaDB query
        break;
    default: throw new \Exception('Unknown database driver');
}
```
This is much more resilient than the string comparison.